url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/de/odysseus/el/tree/impl/ast/AstProperty.java

```diff
@@ -0,0 +1,218 @@
+		Object property = getProperty(bindings, context);
+		if (property == null && strict) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.property.notfound", "null", base));
+		}
+		context.setPropertyResolved(false);
+		Class<?> type = context.getELResolver().getType(context, base, property);
+		if (context.isPropertyResolved()) {
+			if (type != null && (value != null || type.isPrimitive())) {
+				value = bindings.convert(value, type);
+			}
+			context.setPropertyResolved(false);
+		}
+		context.getELResolver().setValue(context, base, property, value);
+		if (!context.isPropertyResolved()) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.property.notfound", property, base));
+		}
+	}
+	
+	protected Method findMethod(String name, Class<?> clazz, Class<?> returnType, Class<?>[] paramTypes) {
+		Method method = null;
+		try {
+			method = clazz.getMethod(name, paramTypes);
+		} catch (NoSuchMethodException e) {
+			throw new MethodNotFoundException(LocalMessages.get("error.property.method.notfound", name, clazz));
+		}
+		method = findAccessibleMethod(method);
+		if (method == null) {
+			throw new MethodNotFoundException(LocalMessages.get("error.property.method.notfound", name, clazz));
+		}
+		if (!ignoreReturnType && returnType != null && !returnType.isAssignableFrom(method.getReturnType())) {
+			throw new MethodNotFoundException(LocalMessages.get("error.property.method.returntype", method.getReturnType(), name, clazz, returnType));
+		}
+		return method;
+	}
+	
+	public MethodInfo getMethodInfo(Bindings bindings, ELContext context, Class<?> returnType, Class<?>[] paramTypes) {
+		Object base = prefix.eval(bindings, context);
+		if (base == null) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.base.null", prefix));
+		}
+		Object property = getProperty(bindings, context);
+		if (property == null && strict) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.method.notfound", "null", base));
+		}
+		String name = bindings.convert(property, String.class);
+		Method method = findMethod(name, base.getClass(), returnType, paramTypes);
+		return new MethodInfo(method.getName(), method.getReturnType(), paramTypes);
+	}
+
+	public Object invoke(Bindings bindings, ELContext context, Class<?> returnType, Class<?>[] paramTypes, Object[] paramValues) {
+		Object base = prefix.eval(bindings, context);
+		if (base == null) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.base.null", prefix));
+		}
+		Object property = getProperty(bindings, context);
+		if (property == null && strict) {
+			throw new PropertyNotFoundException(LocalMessages.get("error.property.method.notfound", "null", base));
+		}
+		String name = bindings.convert(property, String.class);
+		Method method = findMethod(name, base.getClass(), returnType, paramTypes);
+		try {
+			return method.invoke(base, paramValues);
+		} catch (IllegalAccessException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 210.